src/libaktualizr/package_manager/packagemanagerfactory.cc

```diff
@@ -1,55 +1,38 @@
 #include "package_manager/packagemanagerfactory.h"
-#include "package_manager/packagemanagerfake.h"
 
-#ifdef BUILD_OSTREE
-#include "package_manager/ostreemanager.h"
-#ifdef BUILD_DOCKERAPP
-#include "package_manager/dockerappmanager.h"
-#endif
-#endif
+#include "logging/logging.h"
 
-#ifdef BUILD_DEB
-#include "package_manager/debianmanager.h"
-#endif
+static std::list<std::pair<std::string, PackageManagerBuilder>> *registered_pkgms_;
 
-#if defined(ANDROID)
-#include "package_manager/androidmanager.h"
-#endif
+bool PackageManagerFactory::registerPackageManager(const char *name, PackageManagerBuilder builder) {
+  // this trick is needed so that this object is constructed before any element
+  // is added to it
+  static std::list<std::pair<std::string, PackageManagerBuilder>> rpkgms;
 
-#include "logging/logging.h"
+  if (registered_pkgms_ == nullptr) {
+    registered_pkgms_ = &rpkgms;
+  }
+  registered_pkgms_->emplace_back(name, builder);
+  return true;
```

Comment:
Do we need to do something to prevent duplicate names?

Reply:
Yes, it's probably best to fail early in this case.